### PR TITLE
Radio Widget

### PR DESCRIPTION
**Pull request by fullstackctp**
Added Radio Widget functionality. It has validation feature and you can make it according to the groupId as mentioned in issue #759 . This also helps in the issue #224 for supporting radio buttons

### CODE REVIEW

**Review by kmahmood74 — COMMENTED** [review]
Please remove all the files that have formatting changes and nothing else. Fix your IDE to not auto-format files. You only have 2 or so files with real code changes, those should be the only ones in the PR. Remove all else please so we may move forward

**Comment by kmahmood74** [comment]
there are still too many files to check, most are there due to your IDE formatting code. REMOVE those from the commit. Collapse your multiple commits into one please!

**Review by vusters — COMMENTED** [review]
@fullstackctp Please see the comments and make it work

File: lib/widget/input/form_radio.dart [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+class RadioController extends FormFieldController {
+  String groupId = "";
+  String radioValue = "";
+  String? leadingTitle;
+  framework.EnsembleAction? onChange;
+}
+
+class RadioState extends FormFieldWidgetState<RadioWidget> {
+  void onToggle(String groupValue, String groupId) {
```
[the remark this answers is not in the document]
Comment (fullstackctp) [reply lib/widget/input/form_radio.dart:58]: Radio State gets called in createState method of the statefulWidget. The onToggle function was used to experiment with the widget. I don't need it now. I will update the PR to remove any extra code.

**Pull request closed without merging**